core/src/main/python/wlsdeploy/tool/discover/common_resources_discoverer.py

```diff
@@ -112,11 +114,126 @@ def get_datasources(self):
+                if connection_property in properties:
+                    qualified_property_value = properties[connection_property]['Value']
+                    if qualified_property_value:
+                        if qualified_property_value.startswith(WLSDeployArchive.WLSDPLY_ARCHIVE_BINARY_DIR
+                                                     + WLSDeployArchive.ZIP_SEP):
+                            qualified_property_value = os.path.join(self._model_context.get_domain_home()
+                                                                    ,qualified_property_value)
+                        if os.path.exists(qualified_property_value):
+                            fixed_path = self._add_wallet_directory_to_archive(datasource, collected_wallet,
+                                                                               qualified_property_value)
+                            _logger.info('WLSDPLY-06367', connection_property, fixed_path,
+                                         class_name=_class_name, method_name=_method_name)
+                            properties[connection_property]['Value'] = fixed_path
+                        else:
+                            _logger.severe('WLSDPLY-06370', datasource, connection_property,
+                                           properties[connection_property]['Value'])
+                            de = exception_helper.create_discover_exception('WLSDPLY-06370', datasource,
+                                                                            connection_property,
+                                                                            properties[connection_property]['Value'])
+                            _logger.throwing(class_name=_class_name, method_name=_method_name, error=de)
+                            raise de
+
+    def _add_wallet_directory_to_archive(self, datasource, collected_wallet_dictionary, property_value):
+        _method_name = '_add_wallet_directory_to_archive'
+
+        # error if wallet parent path is at oracle home
+        onprem_wallet_parent_path, wallet_name = self._get_wallet_name_and_path(datasource, property_value)
+
+        if self._model_context.is_remote():
+            fixed_path = WLSDeployArchive.getDatabaseWalletArchivePath(wallet_name, property_value)
+            self.add_to_remote_map(property_value, fixed_path,
+                                   WLSDeployArchive.ArchiveEntryType.DB_WALLET.name())
+            return fixed_path
+
+        onprem_wallet_dir_is_not_flat = self._is_on_prem_wallet_dir_flat(onprem_wallet_parent_path, property_value)
+        archive_file = self._model_context.get_archive_file()
+
+        # keep track of the wallet name and from where it is collected
+        #  { <onprem_wallet_parentpath> : { 'wallet_name': <name>, 'path_into_archive': <wallet path into archive>,
+        #      },
+        #     .... }
+        #
+        if not onprem_wallet_parent_path in collected_wallet_dictionary:
```

Comment:
How does the not operator work on a path?  Did you mean to use the `not in` operator? (from Sonar code smells)

Reply:
it's not a path, it is checking item in a dictionary